src/main/java/com/google/sps/servlets/UserFoldersServlet.java

```diff
@@ -21,65 +24,76 @@
 import com.google.sps.data.Folder;
 import java.util.List; 
 import java.util.ArrayList;
+import java.util.Map;
+import java.util.HashMap;
 
-
-/** **/
 @WebServlet("/userfolders")
 public class UserFoldersServlet extends HttpServlet {
-
-  /**
+  
+  private static final String CREATE_FORM_HEADER = "showCreateFormStatus";
+  private static final String USER_FOLDER_HEADER = "userFolders";
+  
+  /*
   * Query all folders from current user
   */
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    
     UserService userService = UserServiceFactory.getUserService();
 
-    if (userService.isUserLoggedIn()) {
-
-      List<Folder> userFolders = new ArrayList<>();
+    Map<String, Object> jsonInfo = new HashMap<>();
+		List<Folder> userFolders = new ArrayList<>();
+    jsonInfo.put(CREATE_FORM_HEADER, false);
 
-      String userKey = request.getParameter("userKey");
+    if (userService.isUserLoggedIn()) {
+      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
 
-      // Query all folders identified by the userKey
-      Query folderQuery = new Query("Folder").setAncestor(KeyFactory.stringToKey(userKey));
+      String userEmail = userService.getCurrentUser().getEmail();
+      Query userQuery = new Query("User").setFilter(new FilterPredicate("email", FilterOperator.EQUAL, userEmail));
+			Entity userEntity = datastore.prepare(userQuery).asSingleEntity();
+      Key userKey = userEntity.getKey();
 
-      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+      Query folderQuery = new Query("Folder").setAncestor(userKey);
       PreparedQuery results = datastore.prepare(folderQuery);
 
-      // Add folders to a list
       if (results != null) {
-        for (Entity entity: results.asIterable()) {
-          Folder folder = Folder.EntityToFolder(entity);
+        for (Entity entity : results.asIterable()) {
+          Folder folder = new Folder(entity);
+          folder.setFolderKey(KeyFactory.keyToString(entity.getKey()));
           userFolders.add(folder);
         }
       }
-      
-      response.setContentType("application/json;");
-      response.getWriter().println(new Gson().toJson(userFolders));
+		
+		jsonInfo.put(CREATE_FORM_HEADER, true);
```

Comment:
does this need to be indented this much?

Reply:
Uhmm that's weird, it seems fine on me IDE...